### PR TITLE
Fix #1379: Handle missing parameters.kapitan.compile gracefully

### LINKED ISSUE #1379: [bug]: Somewhat misleading error msg when `kapitan.compile` params are missing
### Description of the bug

If you are missing a `parameters.kapitan.compile` field in your inventory, running `kapitan compile` results in an error complaining a temporarily directory does not exist.

Inventory examples:
```
parameters:
  kapitan: {}
```

vs

```
parameters:
  kapitan:
    compile: {}
```

The first inventory example results in the error below, the second inventory example just does nothing (as expected):

```
% kapitan --version
0.34.7
```

```
% kapitan c
Rendered inventory (0.01s): discovered 1 targets.
Compiling 1/1 targets using 1 concurrent processes: (14 CPU detected)
Compiled tutorial (0.00s)

Unknown (Non-Kapitan) error occurred:
Traceback (most recent call last):
  File "/opt/venv/lib/python3.11/site-packages/kapitan/targets.py", line 148, in compile_targets
    shutil.copytree(temp_compile_path, compile_path)
  File "/usr/local/lib/python3.11/shutil.py", line 571, in copytree
    with os.scandir(src) as itr:
         ^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpsgly1fzs.kapitan/compiled'


[Errno 2] No such file or directory: '/tmp/tmpsgly1fzs.kapitan/compiled'
```

Actually I forgot to include the class file (`kapitan.common`) containing the kapitan parameters in the target tree, which gives the same error as above.

### Steps to reproduce

```bash
Run kapitan with Docker:

% alias kapitan="docker run -t --rm -v $(pwd):/src:delegated kapicorp/kapitan"
% kapitan --version                                                           
0.34.7


`inventory/classes/applications/microservices.yml`

parameters:
  name: microservices


`inventory/classes/kapitan/common.yml`

parameters:
  target: ${_reclass_:name:full}

  kapitan: {}

  # kapitan:
  #   compile: {}

  # kapitan:
  #   compile:
  #   - output_path: .
  #     input_type: copy
  #     input_paths:
  #       - inventory/classes/applications/microservices.yml
  #   vars:
  #     target: ${target}


`inventory/targets/tutorial.yml`

classes:
  - kapitan.common
  - applications.microservices

# forgetting to include kapitan.common with all the kapitan params
# classes:
#   - applications.microservices
```

### What happened?

Instead of a `FileNotFoundError` traceback, I expected a kapitan compile run which does nothing (no error, no output, no `/src/compiled` dir).

Basically `parameters.kapitan = {}` and `parameters.kapitan.compile = {}` should behave the same imo. For me the `FileNotFoundError` was somewhat misleading, as I was thinking some sort of dir/file permission issue under MacOS and/or Docker.

### Additional context

_No response_

### Version (kapitan)

master

### Version (Python)

I use kapitan directly (pip, docker)

### Python virtual environment

- [ ] pyenv
- [ ] venv
- [x] no environment

### OS information

MacOS with Docker

### Relevant log output

```bash

```

### PR DESCRIPTION
**Pull request by Moep90**
- Skip targets without compile configuration instead of raising FileNotFoundError
- Add validation in load_target_inventory() to check for compile key
- Add safety check before copying temp compile path
- Log informational message when no compilable targets found
- Update coverage requirement from 63% to 64%

Tests:
- Add test for empty parameters.kapitan
- Add test for missing compile key
- Add test for mixed valid/invalid targets

Fixes: https://github.com/kapicorp/kapitan/issues/1379

### CODE REVIEW

File: kapitan/targets.py [hunk 1]
```diff
@@ -212,9 +219,21 @@ def load_target_inventory(inventory, requested_targets, ignore_class_not_found=F
             kapitan_target_configs = target.parameters.kapitan
             # check if parameters.kapitan is empty
             if not kapitan_target_configs:
-                raise InventoryError(
-                    f"InventoryError: {target_name}: parameters.kapitan has no assignment"
+                logger.debug(
+                    f"load_target_inventory: target {target_name} has no kapitan configuration, skipping"
```
[the remark this answers is not in the document]
Comment (Moep90) [reply kapitan/targets.py]: Good point — switched both skip paths from `debug` to `info` (50f05c5). Now skipping a target shows up in normal output, so a forgotten class like `kapitan.common` won't be silently masked:

```
Skipping target <name>: no kapitan configuration (parameters.kapitan is empty)
Skipping target <name>: no kapitan.compile configuration
```

Still resolves #1379 (no more `FileNotFoundError`), but with visible feedback instead of a silent no-op.

**Comment by Moep90** [comment]
Self-review pass after merging master — pushed `8c1cb576`:

**Fixed test regressions (was blocking CI):** the merge with #1502 added `test_targets.py`, whose `MockTarget` has `compile = []`. The new compile-skip dropped those mocks, breaking 5 tests. `MockTarget` now carries a compile config so valid targets load; the old `test_empty_kapitan_raises_inventory_error` is rewritten to assert skip-not-raise (matching this PR's design), plus a missing-compile skip test.

**Restored typo protection:** dropping `raise CompileError("no targets found")` meant `kapitan compile -t typo` exited 0 silently. Re-added an explicit check — an unknown `-t` target now errors, while targets that exist but lack `compile` still no-op per #1379. Covered by a new CLI test.

**Cleanups:** removed a redundant `hasattr(..., "compile")` (it's a guaranteed pydantic field) and an always-true `target_objs` guard after the early return.

Full `test_targets.py` + `test_compile.py` green locally (53 passed). The empty-`kapitan` and empty-`compile` skips are intentional and consistent — happy to revisit if you'd prefer a louder warning over an info log.
